### PR TITLE
Unobsolete the rm802, change its lore, some values and spawn points

### PR DESCRIPTION
**Pull request by KeremBabaG**
#### Summary
Content "restore and change rm 802"

#### Purpose of change
#69603 obsoleted rm802. This aims to restore it, revamp it to make it more like China Lake Launcher, change its lore and hopefully expand Rivtech lore some more.

Also I don't like it when rivtech stuff is deprecated

#### Describe the solution
Restored the rm802 and renamed it as RCI 40x46 pump action grenade launcher. The lore for it goes as follows:
-Rivet and Crimp industries produce a pump action multiple grenade launcher for a navy weapons program and wins it
-enters the military industrial complex in this way, takes first steps to become rivtech
-in 2000s it gets replaced by milkor mgl, phased out as surplus
-bought by police, private militaries and used in labs

I'm planning to add these back into labs and as rare loot in police depts and swat units
Also changed material, length and weight to better resemble irl China Lake Launcher

#### Describe alternatives you've considered
don't do this :(
or change some stuff

#### Testing
shot at children
new descriptions and snippet are also there

#### Additional context
Didn't add spawns yet to get some dev feedback

### CODE REVIEW

File: data/json/items/gun/40x46mm.json [hunk 1]
```diff
@@ -123,6 +123,40 @@
     "pocket_data": [ { "pocket_type": "MAGAZINE", "rigid": true, "ammo_restriction": { "40x46mm": 6 } } ],
     "melee_damage": { "bash": 12 }
   },
+  {
+    "id": "rm802",
+    "copy-from": "launcher_base",
+    "looks_like": "m79",
+    "type": "GUN",
+    "name": { "str": "RCI 40x46 pump action grenade launcher" },
+    "description": "The RCI 40x46 pump action grenade launcher was the first milspec product of the Rivet & Crimp Insustries, Rivtech's predecessor.  Outdoing its competitors in tests, this launcher was adopted in 1981 to replace the aging and inadequate single shot launchers after the Vietnam War.  After it was phased out in favor of Milkor Mgl and became surplus it became the favorite of Riot Control Units and PMCs worldwide.  Its 3+1 capacity lets it maintain a good suppressing fire power and it is lightweight for its size thanks to its aluminum frame.",
```
Comment (TheShadowFerret) [inline data/json/items/gun/40x46mm.json:132]:
```suggestion
    "description": "The RCI 40x46 pump action grenade launcher was the first milspec product of the Rivet & Crimp Industries, Rivtech's predecessor.  Outdoing its competitors in tests, this launcher was adopted in 1981 to replace the aging and inadequate single shot launchers after the Vietnam War.  After it was phased out in favor of Milkor MGL and became surplus it became the favorite of Riot Control Units and PMCs worldwide.  Its 3+1 capacity lets it maintain a good suppressing fire power and it is lightweight for its size thanks to its aluminum frame.",
```
When the RivTech is sus

File: data/json/snippets/newspapers.json [hunk 2]
```diff
@@ -84,7 +84,14 @@
...
+          "str": "FORMER RCI DIVISION PURCHASED BY FOREIGN INVESTORS.  A former division of Rivet and Crimp Industries, a major subcontractor for national defense manufacturers, has recently been purchased by Taiwanese American tech billionaire Henry Gau.  CEO of Formosa OptoTech, Gau is known for his close relationship with Chinese suppliers and manufacturers. Branding the company RivTech, the company will focus on competing in the US Next Generation Squad Weapons trial.  Rivet and Crimp Industries's former weapons manufactory division is previously known for being contracted for the procurement of a pump-action grenade launcher after the Advanced Multiple Rapid Fire Launcher Trial.",
```
Comment (TheShadowFerret) [inline data/json/snippets/newspapers.json:87]:
```suggestion
          "str": "FORMER RCI DIVISION PURCHASED BY FOREIGN INVESTORS.  A former division of Rivet and Crimp Industries, a major subcontractor for national defense manufacturers, has recently been purchased by Taiwanese-American tech billionaire Henry Gau.  CEO of Formosa OptoTech, Gau is known for his close relationship with Chinese suppliers and manufacturers. Branding the company RivTech, the company will focus on competing in the US Next Generation Squad Weapons trial.  Rivet and Crimp Industries' former weapons manufactory division is previously known for being contracted for the procurement of a pump-action grenade launcher after the Advanced Multiple Rapid Fire Launcher Trial.",
```

File: data/json/items/gun/40x46mm.json (hunk 1 above)
Comment (KeremBabaG) [reply data/json/items/gun/40x46mm.json:132]: damn

**Comment by kevingranade** [comment]
The rationales don't hold up.
There's no lore enhancement by putting extraneous details into Rivtech's history, and it doesn't align with the caseless next-generation weapons devrlopment, which is a big part of why the rm802 was removed in the first place.

"Also I don't like it when rivtech stuff is deprecated" isn't doing you any favors,  this is not a valid reason to make the change. 

There is no rationale presented for why this is a good idea, just that it's a gun you want.

**Comment by KeremBabaG** [comment]
lole

**Pull request closed without merging**